mrmustard/lab_dev/transformations/phasenoise.py

```diff
@@ -0,0 +1,89 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The class representing a Phase noise channel.
+"""
+
+from __future__ import annotations
+from typing import Sequence
+from mrmustard import math
+from mrmustard.physics.representations import Representation
+from mrmustard.physics.ansatz.array_ansatz import ArrayAnsatz
+from mrmustard.lab_dev.circuit_components import CircuitComponent
+from .base import Channel
+from ..utils import make_parameter
+import numpy as np
+
+__all__ = ["PhaseNoise"]
+
+
+class PhaseNoise(Channel):
+    r"""
+    The Phase noise channel.
+
+    This class represents the application of a random phase. The distributiuon of the phase
+    is assumed to be a Gaussian with mean zero, and standard deviation `phase_stdev`.
+
+    Args:
+        modes: The modes the channel is applied to
+        phase_stdev: The standard deviation of the random Gaussian noise.```
+
+    ..details::
+        The Fock representation is connected to the Fourier coefficients of the distribution.
+    """
+
+    short_name = "P~"
+
+    def __init__(
+        self,
+        modes: Sequence[int],
+        phase_stdev: float | Sequence[float],
+        phase_stdev_trainable: bool = False,
+        phase_stdev_bounds: tuple[float | None, float | None] = (0.0, None),
+    ):
+        super().__init__(name="PhaseNoise")
+        self._add_parameter(
+            make_parameter(phase_stdev_trainable, phase_stdev, "phase_stdev", phase_stdev_bounds)
+        )
+        self._representation = self.from_ansatz(
+            modes_in=modes, modes_out=modes, ansatz=None
+        ).representation
+
+    def __custom_rrshift__(self, other: CircuitComponent) -> CircuitComponent:
+        r"""
+        Since PhaseNoise admits a particularly nice form in the Fock basis, we have implemented its right-shift operation separately.
+
+        Args:
+            other: the component other than the PhaseNoise object that is present in the contraction
+
+        Output:
+            the result of the contraction.
+        """
+
+        if (len(other.wires.bra) == 0) or (len(other.wires.ket) == 0):
+            return (other @ other.adjoint) >> self
+        array = math.asnumpy(other.fock_array())
+        for mode in self.modes:
+            for count, _ in enumerate(np.nditer(array)):
+                idx = np.zeros(len(array.shape))
+                temp = count
+                for l in range(len(idx)):
+                    idx[-1 - l] = temp % array.shape[-1 - l]
+                    temp = temp // array.shape[-1 - l]
+                array_index = tuple(idx.astype(int))
+                array[array_index] *= np.exp(
+                    -0.5 * (idx[mode] - idx[other.n_modes + mode]) ** 2 * self.phase_stdev.value**2
+                )
```

Comment:
```suggestion
        array = math.asnumpy(other.fock_array())
        mode_indices = np.indices(array.shape)
        for mode in self.modes:
                phase_factors = np.exp(
                -0.5 * (mode_indices[mode] - mode_indices[n_modes + mode])**2 * self.phase_stdev.value**2
        )
        array *= phase_factors
```
I believe this does the same and it should be more efficient.